### PR TITLE
ci: update release names and package details

### PR DESCRIPTION
**Pull request by pjlittle**
Update release package names to use a more standard convention
that includes version. Add more details to `package.json`.

AB#17236

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -47,6 +47,7 @@
         "electron:start:prod": "npm run webpack:prod && npm run electron-start",
         "electron-start": "node src/electron/start",
         "eject": "react-scripts eject",
+        "release-pr": "./scripts/release-pr.sh",
```
Comment (pjlittle) [inline package.json]: 'slightly' easier way to setup a release PR: `$ npm run release-pr --from 2.0.0-preview.1 --to 2.0.0-preview.2`

File: package.json [hunk 2]
```diff
@@ -100,5 +108,9 @@
         "webpack": "^4.19.1",
         "webpack-cli": "^3.1.2",
         "webpack-merge": "^4.1.5"
+    },
+    "engines": {
```
Comment (mydiemho) [inline package.json]: YASS! now we don't have to keep asking what version everyone is running :P

File: electron-builder.yml [hunk 3]
```diff
@@ -4,7 +4,7 @@ directories:
   output: releases
   buildResources: app-icons # this is where app-icons is store
 appId: com.microsoft.vott
-artifactName: '${productName}-${os}.${ext}'
+artifactName: '${productName}-${version}-${platform}.${ext}'
```
Comment (mydiemho) [inline electron-builder.yml]: where is `platform` coming from:

Comment (pjlittle) [reply electron-builder.yml]: it's a standard electron macro: `platform “darwin” | “linux” | “win32” | “mas”`

https://www.electron.build/configuration/configuration#configuration

**Pull request merged**